invenio_circulation/views.py

```diff
@@ -137,3 +141,65 @@ def post(self, pid, record, action, **kwargs):
             pid, record, HTTP_CODES['accepted'],
             links_factory=self.links_factory
         )
+
+
+def build_blueprint_with_state(app):
+    """Item circulation state blueprint."""
+    blueprint = Blueprint(
+        'invenio_circulation_state',
+        __name__,
+        url_prefix='',
+    )
+
+    create_error_handlers(blueprint)
+
+    rec_serializers = {
+        "application/json": (
+            "invenio_records_rest.serializers" ":json_v1_response"
+        )
+    }
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+
+    from invenio_circulation.links import loan_links_factory
+    loan_request = StateResource.as_view(
+        StateResource.view_name, serializers=serializers,
+        ctx=dict(links_factory=loan_links_factory),
+    )
+
+    url = 'circulation/items/<pid_value>/status'
+
+    blueprint.add_url_rule(
+        url, view_func=loan_request, methods=["GET"]
+    )
+    return blueprint
+
+
+class StateResource(ContentNegotiatedMethodView):
+    """Item circulation state resource."""
+
+    view_name = 'state_resource'
+
+    def __init__(self, serializers, ctx, *args, **kwargs):
+        """Resource view contructor."""
+        super(StateResource, self).__init__(serializers, *args, **kwargs)
+        for key, value in ctx.items():
+            setattr(self, key, value)
+
+    def get(self, *args, **kwargs):
+        """Handle GET request for item state."""
+        loans = list(LoansSearch.search_loans_by_pid(
+            item_pid=kwargs.get('pid_value', None),
```

Comment:
What if item_pid is not passed? Do we want to list all `ITEM_ON_LOAN` loans?

Reply:
as I see `item_pid` is string so either it won't match anything for `Loan`, or it will return 404 in case of empty string